### DISCUSSION TITLE
weird whitenoise in the background of audio stream pipewire

### DISCUSSION THREAD

**Discussion by subterfugue** [roc-streaming/roc-toolkit#500] [Q&A] [answered]
I have just discovered roc while surfing matrix rooms and the pipewire module network list. i came across roc and thought that it was really nice and easy to work with overall -- though, there was one issue. there's this annoying weird whitenoise in the background, which is especially bad if i'm listening to music. in the pipewire module there's the `fen.code` option and i had that set to `disabled`, so i'm not sure if that's the issue or if it's something else.

**Comment by subterfugue** [comment, marked as answer]
something else: someone said this could be dithering or something, but they're not sure and i'm also not sure that it is this

**Reply by gavv** [reply]
Hi!

Are you using pipewire for both receiver and sender? 

Can you try to replace them with roc cli tool? Try replacing only receiver, only sender, and both.

This way we can know if the problem specific to pipewire modules.

Regarding cli, see https://roc-streaming.org/toolkit/docs/tools/command_line_tools.html

**Reply by subterfugue** [reply]
Since I use the Roc Droid app on my phone as the receiver, my computer sends the audio stream. Thank you very much for the recommendation, I'll try that out.

**Reply by subterfugue** [reply]
Alright. I've tried to use `roc-send` but it doesn't seem to be working:
```terminal
$ roc-send -vv -i file:./30._Pizzatron_3000.mp3 -s rtp+rs8m://127.0.0.1:10001 -r rs8m://127.0.0.1:10002
14:59:46.472 [dbg] roc_sndio: initializing pulseaudio backend
14:59:46.472 [err] roc_pipeline: parse port: bad IPv4 address: //127.0.0.1
14:59:46.472 [err] roc_send: can't parse remote source port: rtp+rs8m://127.0.0.1:10001
```
This happens with whatever IP I give. 
I think this might be because my roc version is outdated, now that I remebmered to check the version. I'll update and see if this fixes both of my problems.

**Reply by gavv** [reply]
Yes, the log looks like you're using new syntax with old version.

**Reply by gavv** [reply]
Could you also post parameters of your pipewire roc sink?

**Reply by subterfugue** [reply]
Oh, sorry, I forgot to do that. right now i have it set up like https://paste.rs/lr1 to play with the simple pipewire module and also the roc sink. only the roc sink part is relevant, though, so you can ignore the first object that has the name of libpipewire-module-simple or such.

**Reply by subterfugue** [reply]
Alright. The solution was setting `fec.code` to rs8m and restarting pipewire. i'll still attempt to get the roc-toolkit package to be updated on nixos, though. thank you very much for your response and i'm sorry to take up your time.

**Reply by subterfugue** [reply]
Alright. The solution was setting `fec.code` to rs8m and restarting pipewire. i'll still attempt to get the roc-toolkit package to be updated on nixos, though. thank you very much for your response and i'm sorry to take up your time.
